src/coreclr/utilcode/check.cpp

```diff
@@ -127,7 +124,7 @@ void CHECK::Trigger(LPCSTR reason)
         pMessage->AppendASCII(m_condition);
 #endif
 
-        messageString = pMessage->GetANSI(*pScratch);
+        messageString = pMessage->GetUTF8();
```

Comment:
If the message string is Utf8 now, we should use `OutputDebugStringUtf8` below to print it.

Reply:
Also, fix `DbgAssertDialog` to expect Utf8. It ends up hitting `OutputDebugStringA(szExpr)` that has similar problem.